### PR TITLE
fix: use aws default credential chain to load SQS client credentials

### LINKED ISSUE #7: Is it tied to use static AWS_ACCESS_KEY and AWS_SECRET ENV variables only?
According to https://github.com/numaproj-contrib/aws-sqs-source-go/blob/main/main.go#L34-L41

If there's no ENV variables provided, does it still try to use Metadata API to get credentials (this is the approach that those non-credential solutions like KIAM leverage)?

Based on the [doc](https://github.com/aws/aws-sdk-go/blob/main/aws/session/doc.go#L24-L31) from AWS lib, shall we just use `NewSession`?

### PR DESCRIPTION
**Pull request by shubhamdixit863**
closes #7

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,6 +4,9 @@ go 1.21.3
 
 require (
 	github.com/aws/aws-sdk-go v1.46.3
```
Comment (vigith) [inline go.mod]: let's remove the old AWS Go SDK.

**Review by vigith — CHANGES_REQUESTED** [review]

File: pkg/sqs/sqs.go [hunk 2]
```diff
@@ -18,88 +18,91 @@ limitations under the License.
...
 // Read fetches messages from the SQS queue and sends them to the provided message channel.
 func (s *AWSSqsSource) Read(_ context.Context, readRequest sourcesdk.ReadRequest, messageCh chan<- sourcesdk.Message) {
-	var readRequestCount int64 = 10
+	var readRequestCount int32 = 10
 
 	ctx, cancel := context.WithTimeout(context.Background(), readRequest.TimeOut())
 	defer cancel()
 	// If we have un-acked data (data which is received but yet to be deleted from the queue
-	if s.getApproximateMessageCount(ApproximateNumberOfMessagesNotVisible) > 0 {
+	count, err := s.getApproximateMessageCount(approximateNumberOfMessagesNotVisible)
```
Comment (whynowy) [inline pkg/sqs/sqs.go:92]: Does this query return the total invisible message number read by all the clients, or only by the current client? If it's the first case, it means there will be only one client working even there are multiple replicas?

File: main.go [hunk 3]
```diff
@@ -18,36 +18,65 @@ limitations under the License.
...
-	awsSqsSource, err := sqs.NewAWSSqsSource(sqsClient, os.Getenv("AWS_QUEUE"))
+	endPoint := os.Getenv(awsEndpointURL)
+	queueName := os.Getenv(sqsQueueName)
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	client, err := initClient(ctx, endPoint)
+	if err != nil {
+		log.Fatalf("error in creating sqs client %s", err)
+	}
+	queue, err := getQueueUrl(client, queueName, ctx)
```
Comment (KeranYang) [inline main.go]:
```suggestion
	queue, err := client.GetQueueUrl(ctx, &awsSqs.GetQueueUrlInput{QueueName: aws.String(queueName)})
```

`getQueueUrl` is a [shallow](https://medium.com/@alexturdean/shallow-functions-the-silent-killer-of-good-software-527a29ab56c9) function. Please remove it.

File: pkg/sqs/sqs.go [hunk 4]
```diff
@@ -18,88 +18,91 @@ limitations under the License.
...
 
 const (
-	ApproximateNumberOfMessages           = "ApproximateNumberOfMessages"
-	ApproximateNumberOfMessagesNotVisible = "ApproximateNumberOfMessagesNotVisible"
-	MaxNumberOfMessages                   = 10 // MaxNumberOfMessages is messages to return From SQS Valid values: 1 to 10. Default: 1
-	WaitTimeSeconds                       = 20 // WaitTimeSeconds specifies the time (in seconds) the call waits for a message to arrive in the queue before returning. If a message arrives, the call returns early; if not and the time expires, it returns an empty message list.
+	approximateNumberOfMessages           = "ApproximateNumberOfMessages"
+	approximateNumberOfMessagesNotVisible = "ApproximateNumberOfMessagesNotVisible"
+	maxNumberOfMessages                   = 10 // MaxNumberOfMessages is messages to return From SQS Valid values: 1 to 10. Default: 1
+	waitTimeSeconds                       = 20 // WaitTimeSeconds specifies the time (in seconds) the call waits for a message to arrive in the queue before returning. If a message arrives, the call returns early; if not and the time expires, it returns an empty message list.
```
Comment (KeranYang) [inline pkg/sqs/sqs.go]:
```suggestion
	maxNumberOfMessages                   = 10 // maxNumberOfMessages is messages to return From SQS Valid values: 1 to 10. Default: 1
	waitTimeSeconds                       = 20 // waitTimeSeconds specifies the time (in seconds) the call waits for a message to arrive in the queue before returning. If a message arrives, the call returns early; if not and the time expires, it returns an empty message list.
```

File: pkg/sqs/sqs.go [hunk 5]
```diff
@@ -18,88 +18,91 @@ limitations under the License.
...
-		sqsServiceClient: sqsServiceClient,
-	}, nil
+		sqsServiceClient: client,
+		queueURL:         queueURL,
+	}
 }
 
 // GetApproximateMessageCount retrieves the approximate count of messages from an AWS SQS queue based on the given queryType.
-// The queryType can be either "ApproximateNumberOfMessages" or "ApproximateNumberOfMessagesNotVisible".
+// The queryType can be either "approximateNumberOfMessages" or "ApproximateNumberOfMessagesNotVisible".
```
Comment (KeranYang) [inline pkg/sqs/sqs.go]:
```suggestion
// The queryType can be either "approximateNumberOfMessages" or "approximateNumberOfMessagesNotVisible".
```

File: pkg/sqs/sqs.go (hunk 2 above)
Comment (KeranYang) [reply pkg/sqs/sqs.go:92]: +1. According to [aws doc](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/APIReference/API_GetQueueAttributes.html), it seems the first case.

**Review by KeranYang — APPROVED** [review]

**Review by vigith — APPROVED** [review]
please fix the DCO

**Pull request merged**